test/e2e/framework/fixture.go

```
@@ -474,6 +449,64 @@ func (sf SyncerFixture) Start(t *testing.T) *StartedSyncerFixture {
 	// successfully sent a heartbeat to kcp.
 	startedSyncer.WaitForClusterReadyReason(t, ctx, "")
 
+	if useDeployedSyncer {
+		syncerID := syncerConfig.ID()
+
+		// print watch events in syncer namespace. There shouldn't be any.
+		go func() {
+			watch, err := downstreamKubeClient.CoreV1().Events(syncerID).Watch(ctx, metav1.ListOptions{})
+			require.NoError(t, err)
+
+			for watchEvent := range watch.ResultChan() {
+				ev, ok := watchEvent.Object.(*corev1.Event)
+				if !ok {
+					t.Logf("unexpected syncer %q watch event: %v", syncerID, watchEvent)
+					continue
+				}
+				t.Logf("syncer %q namespace event: %s: %s (%s)", syncerID, ev.Type, ev.Message, ev.Reason)
+			}
+		}()
+
+		artifactDir, err := CreateTempDirForTest(t, "artifacts")
+		if err != nil {
+			t.Errorf("failed to create temp dir for syncer artifacts: %v", err)
+		}
+
+		// print logs in syncer namespace.
+		// TODO(sttts): handle pod restarts. But as we assume pods won't die, maybe not necessary.
+		pods, err := downstreamKubeClient.CoreV1().Pods(syncerID).List(ctx, metav1.ListOptions{})
+		if err != nil {
+			t.Errorf("failed to list pods in %s: %v", syncerID, err)
+		}
+		for _, pod := range pods.Items {
+			logs, err := downstreamKubeClient.CoreV1().Pods(syncerID).GetLogs(pod.Name, &corev1.PodLogOptions{Follow: true}).Stream(ctx)
+			if err != nil {
+				t.Errorf("failed to get logs for pod %s in %s: %v", pod.Name, syncerID, err)
+			}
+			r := bufio.NewReader(logs)
+
+			logFilePath := filepath.Join(artifactDir, fmt.Sprintf("syncer-%s-pod-%s.log", syncerID, pod.Name))
+			logFile, err := os.Create(logFilePath)
+			require.NoError(t, err)
+
+			go func() {
+				defer logFile.Close()
```

Are we eating an error here we could log instead?